extensions/azurecore/src/azureResource/utils.ts

```diff
@@ -403,19 +413,23 @@ export async function makeHttpRequest(account: AzureAccount, subscription: azure
 		requestUrl = `${account.properties.providerSettings.settings.armResource.endpoint}${path}`;
 	}
 
-	let response: AxiosResponse | undefined;
+	let response;
 	switch (requestType) {
 		case HttpRequestMethod.GET:
-			response = await axios.get(requestUrl, config);
+			response = await httpClient.sendGetRequestAsync<any>(requestUrl, {
+				headers: reqHeaders
+			});
 			break;
 		case HttpRequestMethod.POST:
-			response = await axios.post(requestUrl, requestBody, config);
+			response = await httpClient.sendPostRequestAsync<HttpClientResponse>(requestUrl, networkRequestOptions);
 			break;
 		case HttpRequestMethod.PUT:
-			response = await axios.put(requestUrl, requestBody, config);
+			response = await httpClient.sendPutRequestAsync<HttpClientResponse>(requestUrl, networkRequestOptions);
 			break;
 		case HttpRequestMethod.DELETE:
-			response = await axios.delete(requestUrl, config);
+			response = await httpClient.sendDeleteRequestAsync<any>(requestUrl, {
```

Comment:
can Patch support be added?  might be needed in the future

Reply:
We can add it later if/when that comes up IMO